### DISCUSSION TITLE
How to bulkinsert a field of datatype.ARRAY from files?

### DISCUSSION THREAD

**Discussion by Wiley-Z** [milvus-io/milvus#34083] [Q&A and General discussion] [answered]
It seems bulkinsert doesn't process ndarray of object type. How should I organize and save these data?

**Comment by xiaofan-luan** [comment, marked as answer]
currently for bulkinsert array you will need to convert numpy to a parquet file. 

here is a code about how to import array
https://github.com/milvus-io/pymilvus/blob/f7a4839a8a6b05620985d25cde47b63247a561e7/examples/example_bulkwriter.py#L51

**Reply by xiaofan-luan** [reply]
current you will have to use string

**Reply by Wiley-Z** [reply]
Thank you! by string you mean varchar, right? I think datatype.STRING has not been implemented yet.

**Reply by yhmo** [reply]
DataType.STRING is reserved, you should use DataType.VARCHAR for string type field.

**Reply by thomcom** [reply]
This example depends on us using minio for storage. How can we do this when we use externalS3 for storage?

**Reply by xiaofan-luan** [reply]
> This example depends on us using minio for storage. How can we do this when we use externalS3 for storage?

I believe S3 and minio has no difference. They share the same client and protocol.
Maybe there are some specific configurations
